diff --git a/gcc/cp/cfns.h b/gcc/cp/cfns.h
index 62cdfab..6218e64 100644
--- a/gcc/cp/cfns.h
+++ b/gcc/cp/cfns.h
@@ -54,6 +54,13 @@ static unsigned int hash (const char *, unsigned int);
 #ifdef __GNUC__
 __inline
 #endif
+
+#ifdef __GNUC__
+__inline
+#ifdef __GNUC_STDC_INLINE__
+__attribute__ ((__gnu_inline__))
+#endif
+#endif
 const char * libc_name_p (const char *, unsigned int);
 /* maximum key range = 391, duplicates = 0 */
 
